src/Oscoin/Data/OscoinTx.hs

```diff
@@ -0,0 +1,333 @@
+        alter (withAccount (deposit bal))  bKey $ ws
+
+    withAccount f v = case v of
+        -- Account exists. Update balance or delete account if zero balance.
+        Just (AccountVal acc) -> case f acc of
+            acc'@Account{..} | accountBalance == 0 -> Nothing
+                             | otherwise           -> Just (AccountVal acc')
+        -- Account doesn't exist. Create it.
+        Nothing -> Just . AccountVal $ (mkAccount b) { accountBalance = bal }
+
+        -- Not an account ...
+        val -> val
+
+-- | Map a handler result into a transaction result.
+handler :: Either Error a -> Either (TxError c) a
+handler (Left err) = Left (ErrHandlerFailed err)
+handler (Right r)  = Right r
+
+-- | Lookup a member in the state.
+lookupMember
+    :: (ByteArrayAccess (Crypto.ShortHash c), CBOR.Serialise (Crypto.ShortHash c))
+    => Address c -> WorldState c -> Either (TxError c) (Member c)
+lookupMember addr ws =
+    case WorldState.lookup key ws of
+        Just (MemberVal m) -> Right m
+        Nothing            -> Left (ErrKeyNotFound key)
+        _                  -> Left (ErrTypeMismatch key)
+  where
+    key = addressKey addr
+
+-- | Lookup a project in the state.
+lookupProject
+    :: (ByteArrayAccess (Crypto.ShortHash c), CBOR.Serialise (Crypto.ShortHash c))
+    => Address c -> WorldState c -> Either (TxError c) (Project c)
+lookupProject addr ws =
+    case WorldState.lookup (addressKey addr) ws of
+        Just (ProjectVal p) -> Right p
+        Nothing             -> Left (ErrKeyNotFound key)
+        _                   -> Left (ErrTypeMismatch key)
+  where
+    key = addressKey addr
+
+-- | Lookup an account in the state.
+lookupAccount
+    :: (ByteArrayAccess (Crypto.ShortHash c), CBOR.Serialise (Crypto.ShortHash c))
+    => Address c -> WorldState c -> Either (TxError c) (Account c)
+lookupAccount addr ws =
+    case WorldState.lookup key ws of
+        Just (AccountVal a) -> Right a
+        Nothing             -> Left (ErrKeyNotFound key)
+        _                   -> Left (ErrTypeMismatch key)
+  where
+    key = addressKey addr
+
+-- | Lookup the current epoch.
+lookupEpoch :: WorldState c -> Either (TxError c) Epoch
+lookupEpoch ws =
+    case WorldState.lookup "epoch" ws of
+        Just (NatVal e) -> Right e
+        Nothing         -> Left (ErrKeyNotFound "epoch")
+        _               -> Left (ErrTypeMismatch "epoch")
+
+-- | The minimum transaction fee that is valid for the given transaction.
+minimumTxFee :: Tx c -> Balance
+minimumTxFee = const 1
+
+-------------------------------------------------------------------------------
+-- TxPayload instances
+-------------------------------------------------------------------------------
+
+deriving instance
+    ( Show (Signature c)
+    , Show (BlockHash c)
+    , Crypto.HasHashing c
+    , Show (PublicKey c)
+    ) => Show (TxPayload c)
+
+deriving instance
+    ( Eq (Signature c)
+    , Eq (BlockHash c)
+    , Eq (PublicKey c)
```

Comment:
Maybe it would be a good idea to add these constraints to the requirements for the `HasDigitalSignature` class like we did for `Hash`.